diff -urp suricata-2.0.9.orig/libhtp/configure.ac suricata-2.0.9/libhtp/configure.ac
--- suricata-2.0.9.orig/libhtp/configure.ac	2015-09-25 10:18:41.000000000 -0400
+++ suricata-2.0.9/libhtp/configure.ac	2015-10-03 08:18:11.755359929 -0400
@@ -3,8 +3,8 @@ dnl ----------------------
 dnl Initialization macros
 dnl ----------------------
 
-AC_INIT([LibHTP], m4_esyscmd([./get-version.sh VERSION]))
-AM_INIT_AUTOMAKE()
+AC_INIT(htp/htp.h, m4_esyscmd([./get-version.sh VERSION]))
+AM_INIT_AUTOMAKE(htp, $PACKAGE_VERSION, no-define)
 
 AC_CONFIG_HEADERS([htp_config_auto_gen.h])
 AC_CONFIG_FILES([htp/htp_version.h])
Only in suricata-2.0.9/libhtp: configure.ac.orig
diff -urp suricata-2.0.9.orig/src/Makefile.am suricata-2.0.9/src/Makefile.am
--- suricata-2.0.9.orig/src/Makefile.am	2015-09-25 10:18:33.000000000 -0400
+++ suricata-2.0.9/src/Makefile.am	2015-10-03 08:18:11.755359929 -0400
@@ -377,7 +377,7 @@ EXTRA_DIST = util-mpm-ac-cuda-kernel.cu
 AM_CPPFLAGS = $(all_includes)
 
 # the library search path.
-suricata_LDFLAGS = $(all_libraries)
+suricata_LDFLAGS = $(all_libraries) -pie
 suricata_LDADD = $(HTP_LDADD)
 
 # Rules to build CUDA ptx modules
@@ -439,7 +439,7 @@ CLEANFILES = $(PTXS) cuda-ptxdump.h
 endif
 
 # default CFLAGS
-AM_CFLAGS = ${OPTIMIZATION_CFLAGS} ${GCC_CFLAGS} ${CLANG_CFLAGS} ${SECCFLAGS} ${PCAP_CFLAGS} -Wall -Wno-unused-parameter -std=gnu99 -DLOCAL_STATE_DIR=\"$(localstatedir)\"
+AM_CFLAGS = ${OPTIMIZATION_CFLAGS} ${GCC_CFLAGS} ${CLANG_CFLAGS} ${SECCFLAGS} ${PCAP_CFLAGS} -Wall -Wno-unused-parameter -std=gnu99 -DLOCAL_STATE_DIR=\"$(localstatedir)\" -DPIE -fPIE
 # different flags for different cases
 if DEBUG
 AM_CFLAGS += -ggdb -O0
Only in suricata-2.0.9/src: Makefile.am.orig
