diff -ur suricata-3.0RC1.orig/src/Makefile.am suricata-3.0RC1/src/Makefile.am
--- suricata-3.0RC1.orig/src/Makefile.am	2015-11-25 13:36:41.000000000 +0100
+++ suricata-3.0RC1/src/Makefile.am	2015-11-26 14:00:04.000000000 +0100
@@ -421,7 +421,7 @@
 AM_CPPFLAGS = $(all_includes)
 
 # the library search path.
-suricata_LDFLAGS = $(all_libraries)
+suricata_LDFLAGS = $(all_libraries) -pie
 suricata_LDADD = $(HTP_LDADD)
 
 # Rules to build CUDA ptx modules
@@ -483,7 +483,7 @@
 endif
 
 # default CFLAGS
-AM_CFLAGS = ${OPTIMIZATION_CFLAGS} ${GCC_CFLAGS} ${CLANG_CFLAGS} ${SECCFLAGS} ${PCAP_CFLAGS} -Wall -Wno-unused-parameter -std=gnu99 -DLOCAL_STATE_DIR=\"$(localstatedir)\"
+AM_CFLAGS = ${OPTIMIZATION_CFLAGS} ${GCC_CFLAGS} ${CLANG_CFLAGS} ${SECCFLAGS} ${PCAP_CFLAGS} -Wall -Wno-unused-parameter -std=gnu99 -DLOCAL_STATE_DIR=\"$(localstatedir)\" -DPIE -fPIE
 # different flags for different cases
 if DEBUG
 AM_CFLAGS += -ggdb -O0
