diff -ur suricata-2.0.orig/doc/Makefile.am suricata-2.0/doc/Makefile.am
--- suricata-2.0.orig/doc/Makefile.am	2014-03-25 04:47:37.000000000 -0400
+++ suricata-2.0/doc/Makefile.am	2014-03-26 19:26:23.661081000 -0400
@@ -30,5 +30,3 @@
 Ubuntu_Installation_from_GIT.txt \
 Windows.txt
 
-docdir = ${datarootdir}/doc/${PACKAGE}
-dist_doc_DATA = ${EXTRA_DIST}
diff -ur suricata-2.0.orig/libhtp/configure.ac suricata-2.0/libhtp/configure.ac
--- suricata-2.0.orig/libhtp/configure.ac	2014-03-25 04:47:54.000000000 -0400
+++ suricata-2.0/libhtp/configure.ac	2014-03-26 19:33:03.982067967 -0400
@@ -3,8 +3,8 @@
 dnl Initialization macros
 dnl ----------------------
 
-AC_INIT([LibHTP], m4_esyscmd([./get-version.sh VERSION]))
-AM_INIT_AUTOMAKE()
+AC_INIT(htp/htp.h, m4_esyscmd([./get-version.sh VERSION]))
+AM_INIT_AUTOMAKE(htp, $PACKAGE_VERSION, no-define)
 
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_FILES([htp/htp_version.h])
diff -ur suricata-2.0.orig/src/Makefile.am suricata-2.0/src/Makefile.am
--- suricata-2.0.orig/src/Makefile.am	2014-03-25 04:47:37.000000000 -0400
+++ suricata-2.0/src/Makefile.am	2014-03-26 19:26:23.662081000 -0400
@@ -375,7 +375,7 @@
 AM_CPPFLAGS = $(all_includes)
 
 # the library search path.
-suricata_LDFLAGS = $(all_libraries)
+suricata_LDFLAGS = $(all_libraries) -pie
 suricata_LDADD = $(HTP_LDADD)
 
 # Rules to build CUDA ptx modules
@@ -437,8 +437,8 @@
 endif
 
 
-#suricata_CFLAGS = -Wall -fno-strict-aliasing 
-AM_CFLAGS = -DLOCAL_STATE_DIR=\"$(localstatedir)\"
+#suricata_CFLAGS = -Wall -fno-strict-aliasing
+AM_CFLAGS = -DLOCAL_STATE_DIR=\"$(localstatedir)\" -DPIE -fPIE
 
 if BUILD_UNITTESTS
 check-am:
